subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/fixture/AbstractJvmFailFastIntegrationSpec.groovy

```diff
@@ -0,0 +1,242 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.fixture
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.hamcrest.Matchers
+import org.junit.Rule
+import spock.lang.Unroll
+
+abstract class AbstractJvmFailFastIntegrationSpec extends AbstractIntegrationSpec {
+    protected static final String FAILED_RESOURCE = "fail"
+    protected static final String OTHER_RESOURCE = "other"
+    protected static final int MAX_WORKERS = 2
+
+    @Rule
+    BlockingHttpServer server = new BlockingHttpServer()
+
+    def setup() {
+        server.start()
+    }
+
+    def "all tests run without fail fast"() {
+        given:
+        withBuildFile()
+        withFailingTest()
+        withNonfailingTest()
+        def testExecution = server.expectConcurrentAndBlock(2, FAILED_RESOURCE, OTHER_RESOURCE)
+
+        when:
+        def gradleHandle = executer.withTasks('test').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        sleep(1000)
+        testExecution.release(OTHER_RESOURCE)
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        result.testClass('pkg.OtherTest').assertTestPassed('passingTest')
+    }
+
+    def "all tests run with --no-fail-fast"() {
+        given:
+        withBuildFile()
+        buildFile << "test { failFast = true }"
+        withFailingTest()
+        withNonfailingTest()
+        def testExecution = server.expectConcurrentAndBlock(2, FAILED_RESOURCE, OTHER_RESOURCE)
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--no-fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        sleep(1000)
+        testExecution.release(OTHER_RESOURCE)
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        result.testClass('pkg.OtherTest').assertTestPassed('passingTest')
+    }
+
+    def "stop test execution with failFast"() {
```

Comment:
The tests that use the task property and the one using the command line flag look almost identical. I am sure we can consolidate them with a `where` clause.

Reply:
done